Scaling adjustment, bisection line added, grid moved to background

valosekj opened the pull request:
Included changes:
- added dynamic axis scaling of individual subplots
- added bisection (diagonal) line
- grid moved to background

I also increased size of linear fit equation, it resulted in conflict with vendor label, so I moved vendor label to lower-right corner.

![fig_t1_t2_agreement_per_vendor](https://user-images.githubusercontent.com/39456460/82757019-1511a580-9dde-11ea-9202-b16cad7f8678.png)


Fixes: https://github.com/sct-pipeline/spine-generic/issues/57

valosekj commented:
Could be done:

![fig_t1_t2_agreement_per_vendor](https://user-images.githubusercontent.com/39456460/82760420-b0f9dc00-9df3-11ea-870e-0d362f337107.png)

jcohenadad commented:
cool! but could you make the grid "squary"? (currently it's a rectangle)

also, can you try putting a box around the y=ax+b? the grid is somewhat distracting

valosekj commented:
Sure! I added box around linear fit equation and changed vendor box's frame color to be same as former. Also, I used `plt.gca().set_aspect("equal")` to enforce grid to be squared.
I also increased space between title and subplots.

![fig_t1_t2_agreement_per_vendor](https://user-images.githubusercontent.com/39456460/82809080-96734180-9e8b-11ea-88fc-208cd19c7e37.png)

valosekj commented on processing/generate_figure.py:
```diff
@@ -603,28 +604,57 @@ def main():
+        frame.set_edgecolor('black')
+        ax.add_artist(legend)
+        # Dynamic scaling of individual subplots based on data
+        offset = 8
```
By changing this, it is possible to "zoom" more to data. But I noticed that not for all numbers is the grid "squared".

jcohenadad commented:
Almost there! I would decrease the font size of the equation (e.g. use the same as the x/y-ticklabel).

Also, can you try to put the equation + R2 in red? just to see if it is more intuitive to interpret the red line (because it is linked with the equation).

Also, i would bring the three plots closer (there is too much unnecessary space between them).

valosekj commented:
Decreased spacing between subplots. Smaller equation font and red color. For me, it looks reasonable to have it in red color.

![fig_t1_t2_agreement_per_vendor](https://user-images.githubusercontent.com/39456460/82907874-7d3fc300-9f67-11ea-8a52-9c0f20c70f94.png)
